mining.go

```diff
@@ -1633,7 +1649,7 @@ mempoolLoop:
 
 			// Retrieve the current top block, whose TxTreeRegular was voted
 			// out.
-			topBlock, err := blockManager.chain.BlockByHash(&prevHash)
+			topBlock, err := g.blockManager.chain.BlockByHash(&prevHash)
```

Comment:
Same thing here regarding the chain.  Basically anything that is a component of block manager should be provided in the `BlkTmplGenerator` struct instead to decouple them.